cmd/cosign/cli/tree.go

```diff
@@ -0,0 +1,120 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/spf13/cobra"
+
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	ociremote "github.com/sigstore/cosign/pkg/oci/remote"
+)
+
+func Tree() *cobra.Command {
+	c := &options.TreeOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "tree",
+		Short:   "Display supply chain security related artifacts for an image such as signatures, SBOMs and attestations",
+		Example: "  cosign tree <IMAGE>",
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return TreeCmd(cmd.Context(), c.Registry, args[0])
+		},
+	}
+
+	c.AddFlags(cmd)
+	return cmd
+}
+
+const (
+	SignatureTagSuffix   = ".sig"
+	SBOMTagSuffix        = ".sbom"
+	AttestationTagSuffix = ".att"
+)
+
+func TreeCmd(ctx context.Context, regOpts options.RegistryOptions, imageRef string) error {
+	var scsaRefs []name.Tag
+	ref, err := name.ParseReference(imageRef)
+	if err != nil {
+		return err
+	}
+
+	remoteOpts := regOpts.GetRegistryClientOpts(ctx)
+
+	fmt.Fprintf(os.Stdout, "📦 Supply Chain Security Related artifacts for an image: %s\n", ref.String())
+
+	sigRef, err := ociremote.SignatureTag(ref, ociremote.WithRemoteOptions(remoteOpts...))
+	if err != nil {
+		return err
+	}
+	_, err = remote.Get(sigRef, remoteOpts...)
+	if err == nil {
+		scsaRefs = append(scsaRefs, sigRef)
+	}
+
+	attRef, err := ociremote.AttestationTag(ref, ociremote.WithRemoteOptions(remoteOpts...))
```

Comment:
Would it work if you have multiple attestations?

Reply:
Umm! When I run cosign download attestation I can see all my attestations. But the cosign tree command still show only one .sig and .att artifacts.

Reply:
I'd expect to see all my attestations and my signature.

Reply:
What should the output of this command look like? What is your suggestion? Should we go one step further to list the layers of this tag?

Reply:
@developer-guy Generally, I'd say listing the layers would be idea. I was expecting to see something like:

```
cosign tree hectorj2f/golang:1.16
📦 Supply Chain Security Related artifacts for an image: hectorj2f/golang:1.16
│
├── 🔐 index.docker.io/hectorj2f/golang:sha256-953dfa4c0a09d03b197f8244a2c33e7fe0077e16add884dc91cf46f9bfe3e442.sig
├── 💾 index.docker.io/hectorj2f/golang:sha256-2e2af677fa07dd70d223fcfc376ccae2ec8e153429c04e60e236c68a1ac71a88.att
├── 💾 index.docker.io/hectorj2f/golang:sha256-bcd0cfee9996c5a8e3f175c8eeff5bd671b071684502f70493c3cdbbddc265b2.att
└── 📝 index.docker.io/hectorj2f/golang:sha256-953dfa4c0a09d03b197f8244a2c33e7fe0077e16add884dc91cf46f9bfe3e442.sbom
```

where I could see on my tree all my attestations. Let me know if you want me to add this code in a separate PR, or you prefer to take care of it.

Reply:
Great, thank you, @hectorj2f. I'll take care of it ASAP.